### PR TITLE
Negate the integration build tag.

### PR DESCRIPTION
**Pull request by elopio**
Chipaca noticed that the lint was not checking integration-tests files, and suggested to reverse the logic of the build negating the tag. This works, and IMO is clearer.

### CODE REVIEW

**Comment by fgimenez** [comment]
I'm getting a FTBFS with this, maybe I'm doing it wrong:

```
$ gbp buildpackage  --git-ignore-branch
gbp:info: Exporting 'HEAD' to '/home/fgimenez/workspace/build-area/ubuntu-snappy-tmp'
gbp:info: Moving '/home/fgimenez/workspace/build-area/ubuntu-snappy-tmp' to '/home/fgimenez/workspace/build-area/ubuntu-snappy-1.6ubuntu1'
 dpkg-buildpackage -rfakeroot -D -us -uc -i -I
dpkg-buildpackage: source package ubuntu-snappy
dpkg-buildpackage: source version 1.6ubuntu1
dpkg-buildpackage: source distribution wily
dpkg-buildpackage: source changed by Michael Vogt <michael.vogt@ubuntu.com>
 dpkg-source -i -I --before-build ubuntu-snappy-1.6ubuntu1
dpkg-buildpackage: host architecture amd64
 fakeroot debian/rules clean
dh clean --buildsystem=golang --with=golang --fail-missing --with systemd
   dh_testdir -O--buildsystem=golang -O--fail-missing
   debian/rules override_dh_auto_clean
make[1]: Entering directory '/home/fgimenez/workspace/build-area/ubuntu-snappy-1.6ubuntu1'
dh_auto_clean -O--buildsystem=golang
rm -vf snappy.8
make[1]: Leaving directory '/home/fgimenez/workspace/build-area/ubuntu-snappy-1.6ubuntu1'
   dh_clean -O--buildsystem=golang -O--fail-missing
 dpkg-source -i -I -b ubuntu-snappy-1.6ubuntu1
dpkg-source: warning: Version number suggests Ubuntu changes, but there is no XSBC-Original-Maintainer field
dpkg-source: info: using source format '3.0 (native)'
dpkg-source: info: building ubuntu-snappy in ubuntu-snappy_1.6ubuntu1.tar.xz
dpkg-source: info: building ubuntu-snappy in ubuntu-snappy_1.6ubuntu1.dsc
 debian/rules build
dh build --buildsystem=golang --with=golang --fail-missing --with systemd
   dh_testdir -O--buildsystem=golang -O--fail-missing
   dh_auto_configure -O--buildsystem=golang -O--fail-missing
   debian/rules override_dh_auto_build
make[1]: Entering directory '/home/fgimenez/workspace/build-area/ubuntu-snappy-1.6ubuntu1'
dh_auto_build
	go install -v github.com/ubuntu-core/snappy/cmd/snapd github.com/ubuntu-core/snappy/cmd/snappy github.com/ubuntu-core/snappy/coreconfig github.com/ubuntu-core/snappy/daemon github.com/ubuntu-core/snappy/dirs github.com/ubuntu-core/snappy/helpers github.com/ubuntu-core/snappy/i18n github.com/ubuntu-core/snappy/i18n/xgettext-go github.com/ubuntu-core/snappy/integration-tests github.com/ubuntu-core/snappy/integration-tests/tests github.com/ubuntu-core/snappy/integration-tests/testutils/autopkgtest github.com/ubuntu-core/snappy/integration-tests/testutils/build github.com/ubuntu-core/snappy/integration-tests/testutils/cli github.com/ubuntu-core/snappy/integration-tests/testutils/common github.com/ubuntu-core/snappy/integration-tests/testutils/config github.com/ubuntu-core/snappy/integration-tests/testutils/data github.com/ubuntu-core/snappy/integration-tests/testutils/image github.com/ubuntu-core/snappy/integration-tests/testutils/partition github.com/ubuntu-core/snappy/integration-tests/testutils/report github.com/ubuntu-core/snappy/integration-tests/testutils/runner github.com/ubuntu-core/snappy/integration-tests/testutils/testutils github.com/ubuntu-core/snappy/integration-tests/testutils/tpl github.com/ubuntu-core/snappy/integration-tests/testutils/wait github.com/ubuntu-core/snappy/lockfile github.com/ubuntu-core/snappy/logger github.com/ubuntu-core/snappy/oauth github.com/ubuntu-core/snappy/partition github.com/ubuntu-core/snappy/pkg github.com/ubuntu-core/snappy/pkg/clickdeb github.com/ubuntu-core/snappy/pkg/lightweight github.com/ubuntu-core/snappy/pkg/remote github.com/ubuntu-core/snappy/pkg/removed github.com/ubuntu-core/snappy/pkg/snapfs github.com/ubuntu-core/snappy/policy github.com/ubuntu-core/snappy/progress github.com/ubuntu-core/snappy/provisioning github.com/ubuntu-core/snappy/release github.com/ubuntu-core/snappy/snappy github.com/ubuntu-core/snappy/systemd
github.com/ubuntu-core/snappy/dirs
github.com/ubuntu-core/snappy/lockfile
golang.org/x/crypto/ssh/terminal
github.com/coreos/go-systemd/activation
github.com/gorilla/context
github.com/ubuntu-core/snappy/logger
github.com/ubuntu-core/snappy/pkg
github.com/cheggaaa/pb
github.com/mvo5/goconfigparser
github.com/ubuntu-core/snappy/pkg/remote
gopkg.in/yaml.v2
github.com/gosexy/gettext
github.com/ubuntu-core/snappy/helpers
github.com/mvo5/uboot-go/uenv
github.com/ubuntu-core/snappy/progress
github.com/blakesmith/ar
github.com/gorilla/mux
github.com/ubuntu-core/snappy/release
gopkg.in/tomb.v2
github.com/jessevdk/go-flags
github.com/peterh/liner
github.com/ubuntu-core/snappy/integration-tests/testutils/testutils
github.com/ubuntu-core/snappy/integration-tests/testutils/tpl
github.com/ubuntu-core/snappy/integration-tests/testutils/autopkgtest
gopkg.in/check.v1
github.com/ubuntu-core/snappy/i18n
github.com/ubuntu-core/snappy/integration-tests/testutils/data
github.com/ubuntu-core/snappy/integration-tests/testutils/config
github.com/ubuntu-core/snappy/integration-tests/testutils/image
github.com/ubuntu-core/snappy/integration-tests/tests
go build github.com/ubuntu-core/snappy/integration-tests/tests: no buildable Go source files in /home/fgimenez/workspace/build-area/ubuntu-snappy-1.6ubuntu1/obj-x86_64-linux-gnu/src/github.com/ubuntu-core/snappy/integration-tests/tests
github.com/ubuntu-core/snappy/integration-tests/testutils/report
github.com/ubuntu-core/snappy/oauth
github.com/ubuntu-core/snappy/pkg/clickdeb
github.com/ubuntu-core/snappy/pkg/snapfs
github.com/ubuntu-core/snappy/policy
github.com/ubuntu-core/snappy/systemd
github.com/ubuntu-core/snappy/integration-tests/testutils/cli
github.com/ubuntu-core/snappy/integration-tests/testutils/runner
github.com/ubuntu-core/snappy/i18n/xgettext-go
github.com/ubuntu-core/snappy/coreconfig
github.com/ubuntu-core/snappy/partition
github.com/ubuntu-core/snappy/provisioning
github.com/ubuntu-core/snappy/integration-tests/testutils/build
github.com/ubuntu-core/snappy/integration-tests/testutils/wait
github.com/ubuntu-core/snappy/integration-tests/testutils/partition
github.com/ubuntu-core/snappy/integration-tests
github.com/ubuntu-core/snappy/integration-tests/testutils/common
github.com/ubuntu-core/snappy/snappy
github.com/ubuntu-core/snappy/pkg/removed
github.com/ubuntu-core/snappy/cmd/snappy
github.com/ubuntu-core/snappy/pkg/lightweight
github.com/ubuntu-core/snappy/daemon
github.com/ubuntu-core/snappy/cmd/snapd
dh_auto_build: go install -v github.com/ubuntu-core/snappy/cmd/snapd github.com/ubuntu-core/snappy/cmd/snappy github.com/ubuntu-core/snappy/coreconfig github.com/ubuntu-core/snappy/daemon github.com/ubuntu-core/snappy/dirs github.com/ubuntu-core/snappy/helpers github.com/ubuntu-core/snappy/i18n github.com/ubuntu-core/snappy/i18n/xgettext-go github.com/ubuntu-core/snappy/integration-tests github.com/ubuntu-core/snappy/integration-tests/tests github.com/ubuntu-core/snappy/integration-tests/testutils/autopkgtest github.com/ubuntu-core/snappy/integration-tests/testutils/build github.com/ubuntu-core/snappy/integration-tests/testutils/cli github.com/ubuntu-core/snappy/integration-tests/testutils/common github.com/ubuntu-core/snappy/integration-tests/testutils/config github.com/ubuntu-core/snappy/integration-tests/testutils/data github.com/ubuntu-core/snappy/integration-tests/testutils/image github.com/ubuntu-core/snappy/integration-tests/testutils/partition github.com/ubuntu-core/snappy/integration-tests/testutils/report github.com/ubuntu-core/snappy/integration-tests/testutils/runner github.com/ubuntu-core/snappy/integration-tests/testutils/testutils github.com/ubuntu-core/snappy/integration-tests/testutils/tpl github.com/ubuntu-core/snappy/integration-tests/testutils/wait github.com/ubuntu-core/snappy/lockfile github.com/ubuntu-core/snappy/logger github.com/ubuntu-core/snappy/oauth github.com/ubuntu-core/snappy/partition github.com/ubuntu-core/snappy/pkg github.com/ubuntu-core/snappy/pkg/clickdeb github.com/ubuntu-core/snappy/pkg/lightweight github.com/ubuntu-core/snappy/pkg/remote github.com/ubuntu-core/snappy/pkg/removed github.com/ubuntu-core/snappy/pkg/snapfs github.com/ubuntu-core/snappy/policy github.com/ubuntu-core/snappy/progress github.com/ubuntu-core/snappy/provisioning github.com/ubuntu-core/snappy/release github.com/ubuntu-core/snappy/snappy github.com/ubuntu-core/snappy/systemd returned exit code 1
debian/rules:84: recipe for target 'override_dh_auto_build' failed
make[1]: *** [override_dh_auto_build] Error 1
make[1]: Leaving directory '/home/fgimenez/workspace/build-area/ubuntu-snappy-1.6ubuntu1'
debian/rules:12: recipe for target 'build' failed
make: *** [build] Error 2
dpkg-buildpackage: error: debian/rules build gave error exit status 2
debuild: fatal error at line 1374:
dpkg-buildpackage -rfakeroot -D -us -uc -i -I failed
gbp:error: 'debuild -i -I' failed: it exited with 29
```

File: integration-tests/testutils/build/build.go [hunk 1]
```diff
@@ -30,7 +30,7 @@ import (
 )
 
 const (
-	buildTestCmd = "go test -tags=integration -c ./integration-tests/tests"
+	buildTestCmd = "go test -c ./integration-tests/tests"
 
 	// IntegrationTestName is the name of the test binary.
 	IntegrationTestName = "integration.test"
```
Comment (fgimenez) [inline integration-tests/testutils/build/build.go]: I think that we should add ```-tags=excludeintegration``` for buildSnappyCliCmd and buildSnapdCliCmd too below

File: run-checks [hunk 2]
```diff
@@ -111,41 +117,32 @@ if [ ! -z "$UNIT" ]; then
...
-    go build -tags=integration integration-tests/main.go
-
     # integration suite in kvm
     if which adt-run >/dev/null 2>&1; then
         echo "Running integration tests on rolling edge"
-        go run -tags=integration integration-tests/main.go --snappy-from-branch
+        go run integration-tests/main.go --snappy-from-branch
         # print the results.
         if which subunit2pyunit >/dev/null 2>&1; then
             subunit-1to2 /tmp/snappy-test/output/artifacts/results.subunit | subunit2pyunit
```
Comment (fgimenez) [inline run-checks]: Another nit, don't know from where exactly but after executing the checks I'm getting a leftover "main" executable in the root of the project, we could get rid of it

**Comment by elopio** [comment]
Thanks for catching that @fgimenez. Please review again.

**Comment by fgimenez** [comment]
Awesome! :+1:

**Comment by elopio** [comment]
This ended being more complex than what I expected, so I think it qualifies for the rule of two reviews. @chipaca, what to take a look?

File: run-checks [hunk 3]
```diff
@@ -111,41 +117,34 @@ if [ ! -z "$UNIT" ]; then
...
-
     echo Building
-    go build -v github.com/ubuntu-core/snappy/...
+    go build -tags=excludeintegration -v github.com/ubuntu-core/snappy/...
 
     # tests
     echo Running tests from $(pwd)
-    for pkg in $(go list ./...); do
-        $goctest -v -coverprofile=.coverage/profile.out $pkg
+    for pkg in $(go list ./... | grep -v integration-tests); do
```
Comment (elopio) [inline run-checks]: Look ogra, I can bash!

File: integration-tests/README.md [hunk 4]
```diff
@@ -53,13 +53,13 @@ You can use this flag to test in a remote machine too.
 With the --filter flag you can select the tests to run. For instance you can
 pass MyTestSuite, MyTestSuite.FirstCustomTest or MyTestSuite.*CustomTest:
```
Comment (chipaca) [inline integration-tests/README.md]: If you quote that with backticks, it'll properly set it as code, and it'll also escape the \* which is messing things up right now. If you don't want to backtick-quote, add a \\ before the \*.

**Comment by chipaca** [comment]
Nothing surprising or untoward in the code. :+1:

File: integration-tests/README.md (hunk 4 above)
Comment (elopio) [inline integration-tests/README.md]: Making a new branch to backtick the README. Thanks for the review, I'll land this.

**Pull request merged**